pkg/frontend/querymiddleware/astmapper/instant_splitting.go

```diff
@@ -0,0 +1,542 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package astmapper
+
+import (
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/prometheus/prometheus/promql/parser"
+)
+
+type instantSplitter struct {
+	splitByInterval        time.Duration
+	embeddedAggregatorExpr *parser.AggregateExpr
```

Comment:
I'm wondering if renaming `embeddedAggregatorExpr` to `outerAggregationExpr` may be more clear. Thoughts?

_As mentioned on Slack, please add a comment to explain what it's used for, cause it's non trivial to understand just looking at the code._

Reply:
Yes, `outerAggregationExpr` seems a better fit. Updated and added the comment 🙂